lua/core/screen.lua

```diff
@@ -300,58 +300,95 @@ Screen.save = function() _norns.screen_save() end
+  ['MULTIPLY'] = 3,
+  ['SCREEN'] = 4,
+  ['OVERLAY'] = 5,
+  ['DARKER'] = 6,
+  ['LIGHT'] = 7,
+  ['COLOR_DODGE'] = 8,
+  ['COLOR_BURN'] = 9,
+  ['HARD_LIGHT'] = 10,
+  ['SOFT_LIGHT'] = 11,
+  ['DIFFERENCE'] = 12,
+  ['EXCLUSION'] = 13,
+  ['CLEAR'] = 14,
+  ['SOURCE'] = 15,
+  ['OVER'] = 16,
+  ['IN'] = 17,
+  ['OUT'] = 18,
+  ['ATOP'] = 19,
+  ['DEST'] = 20,
+  ['DEST_OVER'] = 21,
+  ['DEST_IN'] = 22,
+  ['DEST_OUT'] = 23,
+  ['DEST_ATOP'] = 24,
+  ['SATURATE'] = 25,
+  ['HSL_HUE'] = 26,
+  ['HSL_SATURATION'] = 27,
+  ['HSL_COLOR'] = 28,
+  ['HSL_LUMINOSITY'] = 29,
+}
+
 --- change screen blending mode.
--- @tparam number index blending mode (see list)
+-- @tparam number/string index blending mode (see list), strings are case-insensitive, include '_' between words
 --
 -- more info at https://www.cairographics.org/operators/
 --
--- 0 CLEAR
---
--- 1 SOURCE
---
--- 2 OVER
---
--- 3 IN
---
--- 4 OUT
---
--- 5 ATOP
+-- there are other operators available, see the above link or use tab.print(Screen.BLEND_MODES) in the REPL for the full list.
 --
--- 6 DEST
+-- 0 Over (default)
 --
--- 7 DEST_OVER
+-- 1 XOR: clears any overlapping pixels.
 --
--- 8 DEST_IN
+-- 2 Add: adds together the alpha (brightness) of overlapping pixels.
 --
--- 9 DEST_ATOP
+-- 3 Multiply: multiplies the colors of overlapping pixels, the result is always darker than the two inputs.
 --
--- 10 XOR
+-- 4 Screen: the colors of overlapping pixels are complemented, multiplied, then their product is complimented. the result is always lighter than the two inputs.
 --
--- 11 ADD
+-- 5 Overlay: multiplies colors if destination pixel level is >= 8, screens colors if destination pixel level is < 8.
 --
--- 12 SATURATE
+-- 6 Darken: keeps the darker value of overlapping pixels.
 --
--- 13 MULTIPLY
+-- 7 Lighten: keeps the lighter value of overlapping pixels.
 --
--- 14 SCREEN
+-- 8 Color_Dodge: brightens pixels being drawn over.
 --
--- 15 OVERLAY
+-- 9 Color_Burn: darkens pixels being drawn over.
 --
--- 16 DARKEN
+-- 10 Hard_Light: multiplies colors if source pixel level is >= 8, screens colors if source pixel level is < 8.
 --
--- 17 LIGHTEN
+-- 11 Soft_Light: uses Darken or Lighten depending on the color of the source pixel.
 --
--- 18 COLOR_DODGE
+-- 12 Difference: the result is the absolute value of the difference of the destination and source pixels.
 --
--- 19 COLOR_BURN
---
--- 20 HARD_LIGHT
---
--- 21 SOFT_LIGHT
---
--- 22 DIFFERENCE
---
--- 23 EXCLUSION
-Screen.blend_mode = function(index) _norns.screen_set_operator(index) end
+-- 13 Exclusion: similar to Difference, but has lower contrast.
+-- @usage -- number vs. string input
+-- screen.blend_mode(0)
+-- screen.blend_mode('over')
+-- @usage -- case-insensitivity
+-- screen.blend_mode('hard_light')
+-- screen.blend_mode('hArD_lIgHt')
+-- screen.blend_mode('HARD_LIGHT')
+
+Screen.blend_mode = function(index)
+  if type(index) == "string" then
+    local i = Screen.BLEND_MODES[string.upper(index)]
+    if i ~= nil then
```

Comment:
I wasn't able to test this with norns before pushing the commit, I suspect this would lead to the 'attempt to index nil value' error.